### PR TITLE
Remove Old Mutation Code

### PR DESCRIPTION
**Pull request by GameDungeon**
**Brief Description of What This PR Does**

This removes the old mutation code. I believe that this breaks save compat, unless removing a field is handled somehow. 

**Related Issues**

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: simulation_parameters/SimulationParameters.cs
Comment (hhyyrylainen) [inline simulation_parameters/SimulationParameters.cs:1]: Shouldn't the random chance lists be also removed as if I remember right these were the only uses for those?

File: src/auto-evo/mutations/CommonMutationFunctions.cs [hunk 1]
```diff
@@ -53,6 +55,46 @@ public enum Direction
...
+    public static MicrobeSpecies GenerateRandomSpecies(MicrobeSpecies mutated, MutationWorkMemory workMemory,
+        Random random, float mp = 300)
+    {
+        var mutationStrategy = new AddOrganelleAnywhere(_ => true);
+
+        GameWorld.SetInitialSpeciesProperties(mutated, workMemory.WorkingMemory1, workMemory.WorkingMemory2);
+
+        while (mp > 0)
+        {
+            var mutationList = mutationStrategy.MutationsOf(mutated, mp, true)?.OrderBy(_ => random.Next()).ToList();
```
Comment (hhyyrylainen) [inline src/auto-evo/mutations/CommonMutationFunctions.cs:67]: Wouldn't FirstOrDefault be better here? That avoids potentially building a list of 100 items but then discarding all but the first one.

**Comment by hhyyrylainen** [comment]
This is a good change to do to remove outdated code.

I don't see why this would break save compatibility as this just removes a property which when loaded from a save will be ignored in the future. Or does the removal of a class the JSON loader would want to instantiate cause issues?

**Comment by GameDungeon** [comment]
I didn't know how that was handled so I just put a warning to make sure. I guess I should probably just test it.

**Comment by GameDungeon** [comment]
Ok, I was wrong, everything loads fine

File: simulation_parameters/SimulationParameters.cs
Comment (GameDungeon) [reply simulation_parameters/SimulationParameters.cs:1]: I tried removing those and everything breaks, so I decided to ignore those for now.

Comment (hhyyrylainen) [reply simulation_parameters/SimulationParameters.cs:1]: ??? how does that happen? Wouldn't it be better to figure out why and remove all the outdated code that is no longer needed at once?

Comment (GameDungeon) [reply simulation_parameters/SimulationParameters.cs:1]: When I get home later I can explore it more, but I think the entire organelles simulation instance was null or something like that.

Comment (GameDungeon) [reply simulation_parameters/SimulationParameters.cs:1]: I can no longer reproduce the issues A had before, so I must have changed something else by mistake.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
There's a pretty major issue now: the species generated are random now instead of always the same (as I think was the case previously in the benchmark). This will make benchmark results less consistent than before (not that they were super consistent before).

**Comment by hhyyrylainen** [comment]
For the record I checked that the `random` state is the same before the `GenerateRandomSpecies` method is called. So there has to be something in there that doesn't consistently use random state.

**Comment by GameDungeon** [comment]
AddOrganelleAnywhere's random is not passed in, which is the issued.

**Comment by hhyyrylainen** [comment]
That sounds very plausible. Would you be able to make that change? That would be great for this to then be merged as keeping the benchmark at least a bit more consistent would be excellent.

**Comment by GameDungeon** [comment]
I can in a couple hours from now.

**Comment by hhyyrylainen** [comment]
I'll try to find a good new seed for the benchmark as it now causes many big cells that start dying and playing the damage sound really loudly.

**Review by hhyyrylainen — APPROVED** [review]
I couldn't find an absolutely perfect seed so I've opened up a follow up issue but I'll merge this now as it is much better to have a consistent benchmark species than not have that: https://github.com/Revolutionary-Games/Thrive/issues/5424

**Pull request merged**